### ISSUE TITLE
Facing issue while trying to save the data concurrently for liquibase version 4.22.0

### ISSUE THREAD

**Issue by 06Siddhi** [liquibase/liquibase#4314]
### Search first

- [X] I searched and no similar issues were found

### Description

Facing issue while trying to save the data concurrently like saving data for 30 user thread concurrently at a time, success ratio is 20, 17 or sometimes 15.
Facing this issue on version 4.22.0. I've moved to use Scope.setScopeManager(new ThreadLocalScopeManager()) but now see an error due to scopeId mismatch.

**Below is the error in the logs:
,"Caused By": { "liquibase.exception.LiquibaseException: java.lang.RuntimeException: Cannot end scope wxhajsndor when currently at scope root" : {"Liquibase.java:1610":"liquibase.Liquibase.runInScope","Liquibase.java:229":"liquibase.Liquibase.update","Liquibase.java:213":"liquibase.Liquibase.update,"Caused By": { "java.lang.RuntimeException: Cannot end scope wxhajsndor when currently at scope root" : {"Scope.java:241":"liquibase.Scope.exit","Scope.java:205":"liquibase.Scope.child","Scope.java:193":"liquibase.Scope.child","Scope.java:172":"liquibase.Scope.child","Liquibase.java:1605":"liquibase.Liquibase.runInScope"}}}}**

### Steps To Reproduce

Tried saving data using Jmeter for 30 data in database
I've already set Scope.setScopeManager(new ThreadLocalScopeManager()) inside the try block 

### Expected/Desired Behavior

Should be able to save all the data concurrently in database

### Liquibase Version

4.22.0

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by gombasp** [comment]
Same issue here. Upgraded from 4.3.1 to 4.22.0. We saw the same "cannot end scope..." issues and also the message saying that 4.22.0 has some issues. So we tried with 4.21.1 but the "cannot end scope" problem was still there. In our case it comes when we try to apply patches. As @06Siddhi also noted the Scope.setScopeManager(new ThreadLocalScopeManager()) does not seem to solve the issue. 
App is running on several pods in k8s on GCP. 
Please advise.

**Comment by nvoxland** [comment]
We've been working on improving TreadLocalScopeManager lately. Sorry to hear you're still having problems with it.

Is either of you able to post a stacktrace (they are shown with `--log-level=info` ) ?

**Comment by filipelautert** [comment]
Hello @06Siddhi - I believe PR https://github.com/liquibase/liquibase/pull/4461 may fix this issue. Could you give it a try?

**Comment by filipelautert** [comment]
Hello @06Siddhi ! Could you give my fix a try? You can download the zip file [here](https://github.com/liquibase/liquibase/actions/runs/7984207692/artifacts/1261886648) or obtain it using GPM [(gradle](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-gradle-registry#using-a-published-package) or [maven](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry) ):

```
<dependency>
  <groupId>org.liquibase</groupId>
  <artifactId>liquibase-core</artifactId>
  <version>0977c36eca0546dc20ad02cb99161b13a1eb784b-SNAPSHOT</version>
</dependency>
```

```
api("org.liquibase:liquibase-core:0977c36eca0546dc20ad02cb99161b13a1eb784b-SNAPSHOT")
```

**Comment by MalloD12** [comment]
I think we can close this one. What do you think @filipelautert?

**Comment by filipelautert** [comment]
> I think we can close this one. What do you think @filipelautert?

@MalloD12 agree, as since then we added many improvements to the code related to this issue.
